src/com/oltpbenchmark/DBWorkload.java

```diff
@@ -1018,32 +1018,52 @@ private static Results runWorkload(List<BenchmarkModule> benchList, boolean verb
     LOG.info("Efficiency : " + df.format(efficiency) + "%");
 
     List<List<Integer>> list_latencies = new ArrayList<>();
-    for (int i = 0; i < 5; ++i) {
+    for (int i = 0; i < 6; ++i) {
       list_latencies.add(new ArrayList<Integer>());
     }
+
+    boolean printConnectionLatency =
+      xmlConfig.containsKey("displayHikariPoolConnectionLatency") &&
+      xmlConfig.getBoolean("displayHikariPoolConnectionLatency");
     for (Worker<?> w : workers) {
       for (LatencyRecord.Sample sample : w.getLatencyRecords()) {
         list_latencies.get(sample.tranType - 1).add(sample.operationLatencyUs);
       }
+
+      // Since we have 5 transaction types, use the index 5 to store the latency for the Hikari pool
+      // connections.
+      if (printConnectionLatency) {
+        for (LatencyRecord.Sample sample : w.getConnectionLatencyRecords()) {
+          list_latencies.get(5).add(sample.operationLatencyUs);
```

Comment:
this feels hacky. is there some alternative approach where we can track `overall_latencies` as the original latency semantic (including connection acquisition and all retries), `acq_connection_latencies`as the time taken to acquire a connection, and `final_op_try_latency` as the latency of the final/successful op only (as in excluding retries), and each list can be of size `final int kNumTxnTypes = 5;`?

Reply:
so, going further, each worker can have getLatencyRecords , getConnectionLatencies, and getFinalOpLatencies

Reply:
Changed the code to be easier to read by making the intentions clearer.